src/Middleware/Session/src/DistributedSession.cs

```diff
@@ -368,24 +368,37 @@ private void Serialize(Stream output)
             }
         }
 
-        private void Deserialize(Stream content)
+        private void Deserialize(Span<byte> content)
         {
-            if (content == null || content.ReadByte() != SerializationRevision)
+            var index = 0;
+
+            if (content.IsEmpty || content[index] != SerializationRevision)
             {
                 // Replace the un-readable format.
                 _isModified = true;
                 return;
             }
+            index++;
+
+            var expectedEntries = DeserializeNumFrom3Bytes(content.Slice(index, 3));
```

Comment:
Need to do size check(s) on the `content`. Before we would get -1 from the stream when trying to read a single byte, or 0 when trying to read multiple bytes at a time if it was empty. Now you will get an ArgumentOutOfRange when slicing the Span.